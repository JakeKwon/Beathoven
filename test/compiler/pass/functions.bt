
func printHello() -> unit {
    print(5);
    print("\n");
}

func bar(int i, int j) -> unit {
    i = 3;
    print(i);
    print(" by bar\n");
    foo(j + 2);
}

func foo(int a) -> unit {
    // int a = 1;
    print(a);
    print(" by foo\n");
}

func returnMe(int c) -> int {
    return c;
}

printHello();
printHello();
bar(1,2);
int d = returnMe(5);
print(d);

