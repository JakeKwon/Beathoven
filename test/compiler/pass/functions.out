5
5
3 by bar
4 by foo
5